airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumStateDecoratingIterator.java

```
@@ -0,0 +1,169 @@
+
+    this.syncCheckpointDuration = checkpointDuration;
+    this.syncCheckpointRecords = checkpointRecords;
+    this.isSyncFinished = false;
+    initializeCheckpointValues();
+  }
+
+  /**
+   * Computes the next record retrieved from Source stream. Emits state messages as checkpoints based
+   * on number of records or time lapsed.
+   *
+   * <p>
+   * If this method throws an exception, it will propagate outward to the {@code hasNext} or
+   * {@code next} invocation that invoked this method. Any further attempts to use the iterator will
+   * result in an {@link IllegalStateException}.
+   * </p>
+   *
+   * @return {@link AirbyteStateMessage} containing CDC data or state checkpoint message.
+   */
+  @Override
+  protected AirbyteMessage computeNext() {
+    if (isSyncFinished) {
+      return endOfData();
+    }
+
+    if (sendCheckpointMessage) {
+      AirbyteMessage stateMessage = createStateMessage(checkpointOffset);
+      initializeCheckpointValues();
+      return stateMessage;
+    }
+
+    if (changeEventIterator.hasNext()) {
+      try {
+        ChangeEvent<String, String> event = changeEventIterator.next();
+        recordsLastSync++;
+
+        if (checkpointOffset == null &&
+            (recordsLastSync >= syncCheckpointRecords ||
+                Duration.between(dateTimeLastSync, OffsetDateTime.now()).compareTo(syncCheckpointDuration) > 0)) {
+          checkpointOffset = offsetManager.read();
+        }
+
+        if (checkpointOffset != null
+            && cdcStateHandler.isRecordBehindOffset(checkpointOffset, event)
+            && !cdcStateHandler.isSnapshotEvent(event)
+            && changeEventIterator.hasNext()) {
+          sendCheckpointMessage = true;
+        }
+
+        return DebeziumEventUtils.toAirbyteMessage(event, cdcMetadataInjector, emittedAt);
+      } catch (final Exception e) {
+        LOGGER.error("Message iterator failed to read next record. {}", e.getMessage());
```

We should not swallow exception. 
`throw new RuntimeException(e);`